### PR TITLE
Update README and require TensorFlow>=2.2

### PR DESCRIPTION
**Pull request by st--**
Adds explanation of TensorFlow/TFP dependencies (as brought up by https://github.com/GPflow/GPflow/issues/1623).
Also updates the TensorFlow version requirement to >= 2.2 (which is what TFP 0.10 needs).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,20 +33,21 @@ GPflow would not be the same without the community. **We are grateful to [all co
...
 GPflow implements modern Gaussian process inference for composable kernels and likelihoods. The [online documentation (develop)](http://gpflow.readthedocs.io/en/develop/)/[(master)](http://gpflow.readthedocs.io/en/master/) contains more details.
 
-GPflow 2.0 uses [TensorFlow 2.1+](http://www.tensorflow.org) for running computations, which allows fast execution on GPUs, and uses Python ≥ 3.6.
-
+GPflow 2.1 uses [TensorFlow 2.2+](http://www.tensorflow.org) for running computations, which allows fast execution on GPUs, and uses Python ≥ 3.6.
 
 ## Install GPflow 2
 
-**We have experienced issues with `pip`'s pre-2020 dependency resolver; if you encounter issues with incompatible third-party package versions when installing GPflow using the `pip` commands below, try adding the `--use-feature=2020-resolver` argument.**
+We have experienced issues with `pip`'s pre-2020 dependency resolver; if you encounter issues with incompatible third-party package versions when installing GPflow using the `pip` commands below, try adding the `--use-feature=2020-resolver` argument.
```
Comment (joelberkeley) [inline README.md:40]: btw this is now default pip behaviour

File: README.md [hunk 2]
```diff
@@ -33,20 +33,21 @@ GPflow would not be the same without the community. **We are grateful to [all co
 
 GPflow implements modern Gaussian process inference for composable kernels and likelihoods. The [online documentation (develop)](http://gpflow.readthedocs.io/en/develop/)/[(master)](http://gpflow.readthedocs.io/en/master/) contains more details.
 
-GPflow 2.0 uses [TensorFlow 2.1+](http://www.tensorflow.org) for running computations, which allows fast execution on GPUs, and uses Python ≥ 3.6.
-
+GPflow 2.1 uses [TensorFlow 2.2+](http://www.tensorflow.org) for running computations, which allows fast execution on GPUs, and uses Python ≥ 3.6.
```
Comment (joelberkeley) [inline README.md:36]: gpflow 2.1.3 allows tf 2.1

Comment (st--) [reply README.md:36]: It doesn't nominally require tf 2.2 itself, but it does require tfp 0.10+ which in turn only works with tf 2.2 or larger...

File: README.md (hunk 1 above)
Comment (st--) [reply README.md:40]: Great, thanks! So we can just go and delete this paragraph again?

Comment (joelberkeley) [reply README.md:40]: yup

File: README.md (hunk 2 above)
Comment (joelberkeley) [reply README.md:36]: aha. ok cool

File: README.md (hunk 1 above)
Comment (st--) [reply README.md:40]:
```suggestion
```

**Review by joelberkeley — APPROVED** [review]

**Review by joelberkeley-secondmind — APPROVED** [review]

**Pull request merged**